csharp-basic-topics/AwaitInForEachLoopCsharp/AwaitInForEachLoopCsharp/AwaitInForEachLoop.cs

```diff
@@ -0,0 +1,29 @@
+﻿
+namespace AwaitInForEachLoop
+{
+    public static class AwaitInLoop
+    {
+        public static void PrepNumbers()
+        {
+            Console.WriteLine("Incoming number");
+        }
+        public static async Task<List<int>> ResultAsync()
+        {
+            List<int> numbers = [10, 20, 30, 40, 50, 60, 70, 80, 90];
+
+            foreach (var number in numbers)
+            {
+                PrepNumbers();
+
+                Console.WriteLine($"Processed {number}");
+
+                await Task.Delay(1000);
+
+            }
+            Console.WriteLine("Done Processing");
+
+            return numbers;
```

Comment:
Instead of returning the same number list, I would prefer adding the number in the new list as it finishes and returning that list. That list can be compared in the test case.

Reply:
So you want me to create a new list and assign numbers to it??

Reply:
Yes, so the list will tell us the order in which the processing was completed.